pkg/token/interactive.go

```diff
@@ -0,0 +1,83 @@
+package token
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest/adal"
+)
+
+type InteractiveToken struct {
+	clientID    string
+	resourceID  string
+	tenantID    string
+	oAuthConfig adal.OAuthConfig
+}
+
+// newInteractiveTokenProvider returns a TokenProvider that will fetch a token for the user currently logged into the Interactive.
+// Required arguments include an oAuthConfiguration object and the resourceID (which is used as the scope)
+func newInteractiveTokenProvider(oAuthConfig adal.OAuthConfig, clientID, resourceID, tenantID string) (TokenProvider, error) {
+	if clientID == "" {
+		return nil, errors.New("clientID cannot be empty")
+	}
+	if resourceID == "" {
+		return nil, errors.New("resourceID cannot be empty")
+	}
+	if tenantID == "" {
+		return nil, errors.New("tenantID cannot be empty")
+	}
+
+	return &InteractiveToken{
+		clientID:    clientID,
+		resourceID:  resourceID,
+		tenantID:    tenantID,
+		oAuthConfig: oAuthConfig,
+	}, nil
+}
+
+// Token fetches an azcore.AccessToken from the interactive browser SDK and converts it to an adal.Token for use with kubelogin.
+func (p *InteractiveToken) Token() (adal.Token, error) {
+	emptyToken := adal.Token{}
+
+	// Request a new Interactive token provider
+	authorityFromConfig := p.oAuthConfig.AuthorityEndpoint
+	clientOpts := azcore.ClientOptions{Cloud: cloud.Configuration{
+		ActiveDirectoryAuthorityHost: authorityFromConfig.String(), Services: map[cloud.ServiceName]cloud.ServiceConfiguration{},
```

Comment:
you don't need to specify `Services`

Reply:
OK, will remove Services in next commit.